internal/config/storage.go

```diff
@@ -0,0 +1,177 @@
+package config
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"regexp"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"gitlab.com/distributed_lab/dig"
+	"gitlab.com/distributed_lab/figure/v3"
+	"gitlab.com/distributed_lab/kit/kv"
+)
+
+var DOSpacesURLRegexp = regexp.MustCompile(`^https:\/\/(.+?)\.(.+?)(?:\.cdn)?\.digitaloceanspaces\.com\/(.+)$`)
+
+const maxImageSize = 1 << 22 // 4mb
+
+var (
+	ErrImageTooLarge      = fmt.Errorf("too large image, must be not greater than %d bytes", maxImageSize)
+	ErrIncorrectImageType = fmt.Errorf("incorrect object type, must be image/png or image/jpeg")
+	ErrURLRegexp          = fmt.Errorf("url don't match regexp")
+	ErrBucketNotAllowed   = fmt.Errorf("bucket not allowed")
+)
+
+type Storage struct {
```

Comment:
I suggest we move all of this to the separate package and import it in config. It's just not good to have so much business logic defined in config.